### ISSUE TITLE
Problem training nodes without neighbors of type specified by meta-path.

### ISSUE THREAD

**Issue by sunwback** [shangjingbo1226/ESim#2]
Hi.

If I have nodes like this

```
p1 p
p2 p
c1 c
c2 c
w1 w
w2 w
```

and links

```
p1 c1
p1 w1
p2 w2
p1 p2
w1 w2
c1 c2
```

and paths

```
pcwp 1
pwcp 1
```
the program will give me a segmentation fault.

I guess that is because I specified meta-path 'pcwp' but p1 does not have a neighbor of type 'c'. When handling a 'pcwp' path starts with p1 it may cause a segmentation fault.

**Comment by shangjingbo1226** [comment]
Yeah, I think this could be the reason. I will take a look later. If you already have a fix, please send me a pull request. Thanks!
